src/nwp_consumer/internal/outputs/s3/client.py

```diff
@@ -130,6 +130,14 @@ def listInitTimes(self, *, prefix: pathlib.Path) -> list[dt.datetime]:
     def copyITFolderToCache(self, *, prefix: pathlib.Path, it: dt.datetime) -> list[pathlib.Path]:
         """Overrides the corresponding method in the parent class."""
         initTimeDirPath = self.__bucket / prefix / it.strftime(internal.IT_FOLDER_STRUCTURE_RAW)
+
+        if not self.__fs.exists(initTimeDirPath.as_posix()) or not self.__fs.isdir(initTimeDirPath.as_posix()):
+            log.warn(
+                event="init time folder does not exist in store",
+                path=it.strftime(internal.IT_FOLDER_STRUCTURE_RAW),
+            )
```

Comment:
should you then make that folder?

Reply:
Not at this point - that folder only wants to be made when writing to the store, if it doesn't exist. It doesn't make sense for the copy function to make and then copy an empty folder!